### PR TITLE
fix(css): add color space info to gradient pages

### LINKED ISSUE #30027: linear-gradient() is missing information on <color-interpolation-method>
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/CSS/gradient/linear-gradient

### What specific section or headline is this issue about?

Multiple sections

### What information was incorrect, unhelpful, or incomplete?

There's no information in this article about using the new CSS Color 4 *\<color-interpolation-method\>* syntax to create a linear gradient with interpolations between its colour stops in anything other than the RGB colour space. The only mention of this feature in this article is in the browser compatibility table. It's not even present in the *Formal syntax* section. It is however discussed in the following places elsewhere on MDN but these connections are not necessarily obvious:

* The blog post linked to in the *See also* section.
* The separate articles named *\<color-interpolation-method\>* and *\<hue-interpolation-method\>*.

### What did you expect to see?

I expected to see:
* The fact that interpolation happens within the context of a particular colour space briefly mentioned in the description of what a linear gradient *is* in the *Description* section.
* Inclusion of the relevant syntax in the *Formal syntax* section.
* At least one example contained inside a new subsection within the *Examples* section showing a linear gradient interpolated inside a colour space other than RGB.
* A link to the *\<color-interpolation-method\>* article in the *See also* section.

### Do you have any supporting links, references, or citations?

https://developer.mozilla.org/en-US/blog/css-color-module-level-4/
https://developer.mozilla.org/en-US/docs/Web/CSS/color-interpolation-method
https://developer.mozilla.org/en-US/docs/Web/CSS/hue-interpolation-method

### Do you have anything more you want to share?

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/css/gradient/linear-gradient`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/CSS/gradient/linear-gradient
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/css/gradient/linear-gradient/index.md
* Last commit: https://github.com/mdn/content/commit/5e7d1f9ae2cce0cb3f7693dfb8dc6e8d375b2231
* Document last modified: 2023-07-18T00:16:24.000Z

</details>

### PR DESCRIPTION
**Pull request by OnkarRuikar**
- fixes https://github.com/mdn/content/issues/30027

### CODE REVIEW

File: files/en-us/web/css/gradient/repeating-radial-gradient/index.md [hunk 1]
```diff
@@ -120,6 +120,56 @@ repeating-radial-gradient(farthest-corner at 20% 20%, red 0, green, red 20%);
 
 The elliptical gradient will be centered 20% from the top left, and will repeat 10 times between the center and the farthest corner (the bottom right corner). Browsers supporting multi position color stops will display a red and green striped ellipse. Browsers not supporting the syntax yet will see a gradient that goes from red to black and then from blue to green.
 
+### Interpolation in polor color space with hue interpolation methods
```
Comment (estelle) [inline files/en-us/web/css/gradient/repeating-radial-gradient/index.md:123]: The title is kind of too long; especially if there is no content between the title and the example.

This example doesn't add anything for the person trying to learn about `repeating-radial-gradient`, which is the topic of the page. We can either remove this example from the page, or update it. Take a look at the other examples on the page for examples on how to tweak this.
Also, i think we can hide the HTML.  If we don't, we should the h4 headers.

Comment (teoli2003) [reply files/en-us/web/css/gradient/repeating-radial-gradient/index.md:123]: @OnkarRuikar There is nothing called "polar color space" or "rectangular color space". These are coordinate systems.

It should be "polar coordinates", "rectangular coordinates" ("Cartesian coordinates" is the 3rd kind of coordinate system possible).

One way to work around long titles is to use only _Interpolation with hue interpolation methods_ and to add an introductory sentence to the example.

**Review by teoli2003 — CHANGES_REQUESTED** [review]
Fix "color space" -> "coordinates" where needed.

File: files/en-us/web/css/gradient/repeating-radial-gradient/index.md (hunk 1 above)
Comment (OnkarRuikar) [reply files/en-us/web/css/gradient/repeating-radial-gradient/index.md]: > There is nothing called "polar color space" or "rectangular color space". These are coordinate systems.

Specs do use these terms https://drafts.csswg.org/css-color/#GM-chroma and https://drafts.csswg.org/css-color/#ex-premultiplied-lch

> One way to work around long titles is to use only Interpolation with hue interpolation methods and to add an introductory sentence to the example.

This sounds nice.

**Comment by OnkarRuikar** [comment]
> Fix "color space" -> "coordinates" where needed.

Apart from above mentions MDN also uses these terms https://developer.mozilla.org/en-US/docs/Web/CSS/color-interpolation-method#syntax

File: files/en-us/web/css/gradient/conic-gradient/index.md [hunk 2]
```diff
@@ -215,6 +219,50 @@ div {
 
 {{EmbedLiveSample("Checkerboard", 120, 120)}}
 
+### Interpolation with hue interpolation methods
```
Comment (estelle) [inline files/en-us/web/css/gradient/conic-gradient/index.md:222]:
```suggestion
### Hue interpolation methods
```

or "Interpolating with hue"

including interpolation twice in a header seems excessive

File: files/en-us/web/css/gradient/conic-gradient/index.md [hunk 3]
```diff
@@ -215,6 +219,50 @@ div {
...
+}
+
+.longer {
+  background-image: conic-gradient(in hsl longer hue, red, blue);
+}
+```
+
+The box on the left uses [shorter interpolation](/en-US/docs/Web/CSS/hue-interpolation-method#shorter) so color goes straight from red to blue taking short arc on [color wheel](/en-US/docs/Glossary/Color_wheel). And the box on the right uses [longer interpolation](/en-US/docs/Web/CSS/hue-interpolation-method#longer) so the color goes from red to blue via green taking longer arc.
+
+{{EmbedLiveSample('Interpolation with hue interpolation methods', 240, 200)}}
```
Comment (estelle) [inline files/en-us/web/css/gradient/conic-gradient/index.md:264]: if changed, change this too.

File: files/en-us/web/css/gradient/conic-gradient/index.md [hunk 4]
```diff
@@ -215,6 +219,50 @@ div {
...
+.shorter {
+  background-image: conic-gradient(in hsl shorter hue, red, blue);
+}
+
+.longer {
+  background-image: conic-gradient(in hsl longer hue, red, blue);
+}
+```
+
+The box on the left uses [shorter interpolation](/en-US/docs/Web/CSS/hue-interpolation-method#shorter) so color goes straight from red to blue taking short arc on [color wheel](/en-US/docs/Glossary/Color_wheel). And the box on the right uses [longer interpolation](/en-US/docs/Web/CSS/hue-interpolation-method#longer) so the color goes from red to blue via green taking longer arc.
```
Comment (estelle) [inline files/en-us/web/css/gradient/conic-gradient/index.md:262]:
```suggestion
The box on the left uses [shorter interpolation](/en-US/docs/Web/CSS/hue-interpolation-method#shorter), meaning color goes straight from red to blue using the shorter arc on [color wheel](/en-US/docs/Glossary/Color_wheel). The box on the right uses [longer interpolation](/en-US/docs/Web/CSS/hue-interpolation-method#longer), meaning the color goes from red to blue using the longer arc, traversing through greens, yellows, and oranges.
```

File: files/en-us/web/css/gradient/linear-gradient/index.md [hunk 5]
```diff
@@ -147,6 +157,54 @@ body {
...
+
+```css hidden
+div {
+  height: 50vh;
+  color: white;
+  font-weight: bolder;
+}
+```
+
+In this example for interpolation hsl color system is being used and hue is being interpolated.
```
Comment (estelle) [inline files/en-us/web/css/gradient/linear-gradient/index.md:192]:
```suggestion
In this example, the hsl color system is being used and hue is being interpolated.
```
link hsl and hue

File: files/en-us/web/css/gradient/linear-gradient/index.md [hunk 6]
```diff
@@ -147,6 +157,54 @@ body {
...
+.shorter {
+  background: linear-gradient(90deg in hsl shorter hue, red, blue);
+}
+
+.longer {
+  background: linear-gradient(90deg in hsl longer hue, red, blue);
+}
+```
+
+The box on the top uses [shorter interpolation](/en-US/docs/Web/CSS/hue-interpolation-method#shorter) so color goes straight from red to blue taking short arc on [color wheel](/en-US/docs/Glossary/Color_wheel). And the box on the bottom uses [longer interpolation](/en-US/docs/Web/CSS/hue-interpolation-method#longer) so the color goes from red to blue via green taking longer arc.
```
Comment (estelle) [inline files/en-us/web/css/gradient/linear-gradient/index.md:204]: same edits as the example above. Do you want to use different colors so we have two slightly different short and long examples?

File: files/en-us/web/css/gradient/linear-gradient/index.md [hunk 7]
```diff
@@ -147,6 +157,54 @@ body {
...
+
+```css
+body {
+  background: linear-gradient(90deg in oklab, blue, red);
+}
+```
+
+{{EmbedLiveSample("Interpolation in rectangular color space", 120, 120)}}
+
+### Interpolation with hue interpolation methods
```
Comment (estelle) [inline files/en-us/web/css/gradient/linear-gradient/index.md:177]: same edit

File: files/en-us/web/css/gradient/radial-gradient/index.md [hunk 8]
```diff
@@ -121,6 +124,58 @@ Color-stop points are positioned on a _virtual gradient ray_ that extends horizo
 
 {{EmbedLiveSample('Non-centered_gradient', 240, 120)}}
 
+### Interpolation with hue interpolation methods
```
Comment (estelle) [inline files/en-us/web/css/gradient/radial-gradient/index.md:127]: same edits as above

**Review by estelle — CHANGES_REQUESTED** [review]
made some suggestions on the first example, which are valid for all other examples. Consider changing the colors for each example so we have different examples of long and short color arcs.

File: files/en-us/web/css/gradient/conic-gradient/index.md (hunk 2 above)
Comment (OnkarRuikar) [reply files/en-us/web/css/gradient/conic-gradient/index.md:222]: Using "Interpolating with hue".

File: files/en-us/web/css/gradient/conic-gradient/index.md (hunk 3 above)
Comment (OnkarRuikar) [reply files/en-us/web/css/gradient/conic-gradient/index.md:264]: done

File: files/en-us/web/css/gradient/conic-gradient/index.md (hunk 4 above)
Comment (OnkarRuikar) [reply files/en-us/web/css/gradient/conic-gradient/index.md:262]: done

File: files/en-us/web/css/gradient/linear-gradient/index.md (hunk 5 above)
Comment (OnkarRuikar) [reply files/en-us/web/css/gradient/linear-gradient/index.md:192]: done

File: files/en-us/web/css/gradient/linear-gradient/index.md (hunk 6 above)
Comment (OnkarRuikar) [reply files/en-us/web/css/gradient/linear-gradient/index.md:204]: Updated all the docs with above review comments. Examples with different colors would've made sense if the examples were on one page in same section. On separate pages readers won't even notice.

File: files/en-us/web/css/gradient/radial-gradient/index.md (hunk 8 above)
Comment (OnkarRuikar) [reply files/en-us/web/css/gradient/radial-gradient/index.md:127]: Updated all the pages with above review comments.

File: files/en-us/web/css/gradient/conic-gradient/index.md [hunk 9]
```diff
@@ -215,6 +219,50 @@ div {
...
+.shorter {
+  background-image: conic-gradient(in hsl shorter hue, red, blue);
+}
+
+.longer {
+  background-image: conic-gradient(in hsl longer hue, red, blue);
+}
+```
+
+The box on the left uses [shorter interpolation](/en-US/docs/Web/CSS/hue-interpolation-method#shorter), meaning color goes straight from red to blue using the shorter arc on [color wheel](/en-US/docs/Glossary/Color_wheel). The box on the right uses [longer interpolation](/en-US/docs/Web/CSS/hue-interpolation-method#longer), meaning the color goes from red to blue using the longer arc, traversing through greens, yellows, and oranges.
```
Comment (estelle) [inline files/en-us/web/css/gradient/conic-gradient/index.md:262]:
```suggestion
The box on the left uses [shorter interpolation](/en-US/docs/Web/CSS/hue-interpolation-method#shorter), meaning color goes straight from red to blue using the shorter arc on the [color wheel](/en-US/docs/Glossary/Color_wheel). The box on the right uses [longer interpolation](/en-US/docs/Web/CSS/hue-interpolation-method#longer), meaning the color goes from red to blue using the longer arc, traversing through greens, yellows, and oranges.
```

File: files/en-us/web/css/gradient/linear-gradient/index.md [hunk 10]
```diff
@@ -147,6 +157,54 @@ body {
...
+
+```css hidden
+div {
+  height: 50vh;
+  color: white;
+  font-weight: bolder;
+}
+```
+
+In this example for interpolation [hsl](/en-US/docs/Web/CSS/color_value/hsl) color system is being used and [hue](/en-US/docs/Web/CSS/hue) is being interpolated.
```
Comment (estelle) [inline files/en-us/web/css/gradient/linear-gradient/index.md:192]:
```suggestion
In this example for interpolation, [hsl](/en-US/docs/Web/CSS/color_value/hsl) color system is being used and [hue](/en-US/docs/Web/CSS/hue) is being interpolated.
```

File: files/en-us/web/css/gradient/radial-gradient/index.md [hunk 11]
```diff
@@ -121,6 +124,58 @@ Color-stop points are positioned on a _virtual gradient ray_ that extends horizo
...
+}
+
+.longer::before {
+  content: "longer hue";
+  display: block;
+  margin-top: -1rem;
+}
+```
+
+In this example for interpolation [hsl](/en-US/docs/Web/CSS/color_value/hsl) color system is being used and [hue](/en-US/docs/Web/CSS/hue) is being interpolated.
```
Comment (estelle) [inline files/en-us/web/css/gradient/radial-gradient/index.md:155]:
```suggestion
In this example for interpolation, [hsl](/en-US/docs/Web/CSS/color_value/hsl) color system is being used and [hue](/en-US/docs/Web/CSS/hue) is being interpolated.
```

File: files/en-us/web/css/gradient/radial-gradient/index.md [hunk 12]
```diff
@@ -121,6 +124,58 @@ Color-stop points are positioned on a _virtual gradient ray_ that extends horizo
...
+.longer {
+  background-image: radial-gradient(
+    circle at center in hsl longer hue,
+    red,
+    blue
+  );
+}
+```
+
+The box on the left uses [shorter interpolation](/en-US/docs/Web/CSS/hue-interpolation-method#shorter), meaning color goes straight from red to blue using the shorter arc on [color wheel](/en-US/docs/Glossary/Color_wheel). The box on the right uses [longer interpolation](/en-US/docs/Web/CSS/hue-interpolation-method#longer), meaning the color goes from red to blue using the longer arc, traversing through greens, yellows, and oranges.
```
Comment (estelle) [inline files/en-us/web/css/gradient/radial-gradient/index.md:175]:
```suggestion
The box on the left uses [shorter interpolation](/en-US/docs/Web/CSS/hue-interpolation-method#shorter), meaning the color goes straight from red to blue using the shorter arc on [color wheel](/en-US/docs/Glossary/Color_wheel). The box on the right uses [longer interpolation](/en-US/docs/Web/CSS/hue-interpolation-method#longer), meaning the color goes from red to blue using the longer arc, traversing through greens, yellows, and oranges.
```

File: files/en-us/web/css/gradient/repeating-conic-gradient/index.md [hunk 13]
```diff
@@ -157,6 +161,58 @@ div {
...
+}
+
+.longer::before {
+  content: "longer hue";
+  display: block;
+  margin-top: -1rem;
+}
+```
+
+In this example for interpolation [hsl](/en-US/docs/Web/CSS/color_value/hsl) color system is being used and [hue](/en-US/docs/Web/CSS/hue) is being interpolated.
```
Comment (estelle) [inline files/en-us/web/css/gradient/repeating-conic-gradient/index.md:192]:
```suggestion
In this example for interpolation, [hsl](/en-US/docs/Web/CSS/color_value/hsl) color system is being used and [hue](/en-US/docs/Web/CSS/hue) is being interpolated.
```

File: files/en-us/web/css/gradient/repeating-conic-gradient/index.md [hunk 14]
```diff
@@ -157,6 +161,58 @@ div {
...
+.longer {
+  background-image: repeating-conic-gradient(
+    in hsl longer hue,
+    red,
+    blue 180deg
+  );
+}
+```
+
+The box on the left uses [shorter interpolation](/en-US/docs/Web/CSS/hue-interpolation-method#shorter), meaning color goes straight from red to blue using the shorter arc on [color wheel](/en-US/docs/Glossary/Color_wheel). The box on the right uses [longer interpolation](/en-US/docs/Web/CSS/hue-interpolation-method#longer), meaning the color goes from red to blue using the longer arc, traversing through greens, yellows, and oranges.
```
Comment (estelle) [inline files/en-us/web/css/gradient/repeating-conic-gradient/index.md:212]:
```suggestion
The box on the left uses [shorter interpolation](/en-US/docs/Web/CSS/hue-interpolation-method#shorter), meaning color goes straight from red to blue using the shorter arc on the [color wheel](/en-US/docs/Glossary/Color_wheel). The box on the right uses [longer interpolation](/en-US/docs/Web/CSS/hue-interpolation-method#longer), meaning the color goes from red to blue using the longer arc, traversing through greens, yellows, and oranges.
```

File: files/en-us/web/css/gradient/repeating-linear-gradient/index.md [hunk 15]
```diff
@@ -115,6 +125,63 @@ body {
...
+```css hidden
+div {
+  height: 50vh;
+  color: #330;
+  font-weight: bolder;
+  padding-left: 1.5rem;
+}
+```
+
+In this example for interpolation [hsl](/en-US/docs/Web/CSS/color_value/hsl) color system is being used and [hue](/en-US/docs/Web/CSS/hue) is being interpolated.
```
Comment (estelle) [inline files/en-us/web/css/gradient/repeating-linear-gradient/index.md:161]:
```suggestion
In this example for interpolation, [hsl](/en-US/docs/Web/CSS/color_value/hsl) color system is being used and [hue](/en-US/docs/Web/CSS/hue) is being interpolated.
```

File: files/en-us/web/css/gradient/repeating-linear-gradient/index.md [hunk 16]
```diff
@@ -115,6 +125,63 @@ body {
...
+.longer {
+  background: repeating-linear-gradient(
+    90deg in hsl longer hue,
+    red,
+    blue 300px
+  );
+}
+```
+
+The box on the top uses [shorter interpolation](/en-US/docs/Web/CSS/hue-interpolation-method#shorter), meaning color goes straight from red to blue using the shorter arc on [color wheel](/en-US/docs/Glossary/Color_wheel). The box on the bottom uses [longer interpolation](/en-US/docs/Web/CSS/hue-interpolation-method#longer), meaning the color goes from red to blue using the longer arc, traversing through greens, yellows, and oranges.
```
Comment (estelle) [inline files/en-us/web/css/gradient/repeating-linear-gradient/index.md:181]:
```suggestion
The box on the top uses [shorter interpolation](/en-US/docs/Web/CSS/hue-interpolation-method#shorter), meaning the color goes from red to blue using the shorter arc on the [color wheel](/en-US/docs/Glossary/Color_wheel). The box on the bottom uses [longer interpolation](/en-US/docs/Web/CSS/hue-interpolation-method#longer), meaning the color goes from red to blue using the longer arc, traversing through greens, yellows, and oranges.
```

File: files/en-us/web/css/gradient/repeating-radial-gradient/index.md [hunk 17]
```diff
@@ -120,6 +120,58 @@ repeating-radial-gradient(farthest-corner at 20% 20%, red 0, green, red 20%);
...
+}
+
+.longer::before {
+  content: "longer hue";
+  display: block;
+  margin-top: -1rem;
+}
+```
+
+In this example for interpolation [hsl](/en-US/docs/Web/CSS/color_value/hsl) color system is being used and [hue](/en-US/docs/Web/CSS/hue) is being interpolated.
```
Comment (estelle) [inline files/en-us/web/css/gradient/repeating-radial-gradient/index.md:151]:
```suggestion
In this example for interpolation, [hsl](/en-US/docs/Web/CSS/color_value/hsl) color system is being used and [hue](/en-US/docs/Web/CSS/hue) is being interpolated.
```

File: files/en-us/web/css/gradient/repeating-radial-gradient/index.md [hunk 18]
```diff
@@ -120,6 +120,58 @@ repeating-radial-gradient(farthest-corner at 20% 20%, red 0, green, red 20%);
...
+.longer {
+  background-image: repeating-radial-gradient(
+    circle at center in hsl longer hue,
+    red 30px,
+    blue 60px
+  );
+}
+```
+
+The box on the left uses [shorter interpolation](/en-US/docs/Web/CSS/hue-interpolation-method#shorter), meaning color goes straight from red to blue using the shorter arc on [color wheel](/en-US/docs/Glossary/Color_wheel). The box on the right uses [longer interpolation](/en-US/docs/Web/CSS/hue-interpolation-method#longer), meaning the color goes from red to blue using the longer arc, traversing through greens, yellows, and oranges.
```
Comment (estelle) [inline files/en-us/web/css/gradient/repeating-radial-gradient/index.md:171]:
```suggestion
The box on the left uses [shorter interpolation](/en-US/docs/Web/CSS/hue-interpolation-method#shorter), meaning the color goes from red to blue using the shorter arc on [color wheel](/en-US/docs/Glossary/Color_wheel). The box on the right uses [longer interpolation](/en-US/docs/Web/CSS/hue-interpolation-method#longer), meaning the color goes from red to blue using the longer arc, traversing through greens, yellows, and oranges.
```

**Review by estelle — APPROVED** [review]
Thanks!

**Review by estelle — APPROVED** [review]

**Review by teoli2003 — APPROVED** [review]
👍

**Pull request merged**